client/js/socket-events/msg.js

```diff
@@ -47,4 +50,17 @@ socket.on("msg", function(data) {
 			.find(".unread-marker")
 			.appendTo(container);
 	}
+
+	// Message arrived in a non active channel, trim it to 100 messages
+	if (activeChannelId !== target && container.find(".msg").slice(0, -100).remove().length) {
+		channel.find(".show-more").addClass("show");
+
+		// Remove date-seperators that would otherwise
+		// be "stuck" at the top of the channel
+		channel.find(".date-marker-container").each(function() {
+			if ($(this).next().hasClass("date-marker-container")) {
+				$(this).remove();
+			}
+		});
+	}
```

Comment:
Is it worth putting this in a debounce function so it doesn't fire more than once every N seconds (10?)? I'm concerned this could be called a lot in very active channels.